src/directives/markers.js

```diff
@@ -269,6 +269,9 @@ angular.module('ui-leaflet').directive('markers',
                             return;
                         _create(newMarkers, oldMarkers);
                     });
+                    scope.$on('$destroy', function () {
+                        _destroy(leafletScope.markers, {}, leafletMarkers, map, layers);
```

Comment:
Yeap, but if there is some cleaning to be done before removing the marker, then we might need to call the { `deleteMarker` explicitly...